examples/playground/src/widgets/QuizWidget.tsx

```diff
@@ -0,0 +1,27 @@
+import type { WidgetProps } from '@rjsf/core';
+import { useState } from 'react';
+
+export default function QuizWidget(props: Partial<WidgetProps>) {
+  const [answers, setAnswers] = useState<string[]>([]);
+
+  function setAnswerStatus(answer: string) {
+    const newAnswers =
+      answers.findIndex((value) => value === answer) === -1
+        ? [...answers, answer]
+        : answers.filter((e) => e !== answer);
+    setAnswers(newAnswers);
+    props.onChange && props.onChange(newAnswers);
+  }
+
+  return (
+    <>
+      Quiz
+      <label>{props.label}</label>
+      <input type={'checkbox'} onChange={() => setAnswerStatus('a')} />
+      reponse a
+      <input type={'checkbox'} onChange={() => setAnswerStatus('b')} /> reponse b
+      <input type={'checkbox'} onChange={() => setAnswerStatus('c')} /> reponse c
+      <input type={'checkbox'} onChange={() => setAnswerStatus('d')} /> reponse d
+    </>
+  );
+}
```

Comment:
```suggestion
import { ChangeEvent, useState } from 'react';
import type { WidgetProps } from '@rjsf/core';

export default function QuizWidget(props: Partial<WidgetProps>) {
  const choices = ['a', 'b', 'c', 'd'] as const;
  const [answers, setAnswers] = useState<Partial<typeof choices[number]>[]>([]);

  function handleChange(event: ChangeEvent<HTMLInputElement>) {
    const answer = event.target.value as typeof choices[number];
    const checked = event.target.checked;
    const newAnswers = checked ? [...answers, answer] : answers.filter((e) => e !== answer);

    setAnswers(newAnswers);
    props.onChange && props.onChange(newAnswers);
  }

  return (
    <>
      <h3>Quiz {props.label}</h3>
      {choices.map((choice) => (
        <label key={choice}>
          <input type="checkbox" value={choice} onChange={handleChange} />
          Answer a
        </label>
      ))}
    </>
  );
}
```

Reply:
I updated with your code, but i'll modify it further later so it works well (still conflicting with array vs string types...